csharp/src/Microsoft.ML.OnnxRuntime/InferenceSession.shared.cs

```diff
@@ -1001,44 +1023,56 @@ private string GetOutputName(ulong index, out byte[] utf8)
                                            allocator.Pointer,
                                            out IntPtr nameHandle));
 
-            using (var ortAllocation = new OrtMemoryAllocation(allocator, nameHandle, 0))
+            try
             {
                 NativeOnnxValueHelper.StringAndUtf8FromNative(nameHandle, out str, out utf8);
             }
+            finally
+            {
+                allocator.FreeMemory(nameHandle);
+            }
```

Comment:
It is not always the case that the memory comes from the allocator.

Reply:
If StringAndUtf8FromNative allowed passing in an optional allocator it could handle the release and simplify a lot of calling sites. It feels somewhat opaque in the current setup how/when the handle should be freed, and I expect doing so will be overlooked.

Reply:
I reworked the function